homework-submission/week4-class16/src/components/UpdateTodo.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import moment from "moment";
+import "react-datepicker/dist/react-datepicker.css";
+import Form from "./Form";
+import { observer, inject } from "mobx-react";
+
+@inject("TodosStore")
+@observer
+class UpdateTodo extends Component {
+  state = { description: "", deadline: moment() };
+
+  handleDescriptionUpdate = e => {
+    this.setState({
+      description: e.target.value
+    });
+  };
+
+  handleDeadlineUpdate = date => {
+    this.setState({ deadline: date });
+  };
+  handleSubmit = (todoId, event) => {
```

Comment:
This method have to be in the store as an action.

So you will only pass from here the state, for making things organized, means: store have the actions, component have the state.

Reply:
you mean handleSubmit, right? actually, this function calls in the next line handleUpdate function which indeed an action in the store.